tests/main/snaps-state/task.yaml

```diff
@@ -74,3 +74,58 @@ execute: |
         snap remove "$SNAP_JAILMODE"
         rm -f "$TESTSLIB/snaps/${SNAP_JAILMODE}/${SNAP_JAILMODE}_1.0_all.snap"
     fi
+
+    # check the command is-confinement-supported works
+    case "$SPREAD_SYSTEM" in
+        fedora-*)
+            not "$TESTSTOOLS"/snaps-state is-confinement-supported classic
+            not "$TESTSTOOLS"/snaps-state is-confinement-supported strict
+            "$TESTSTOOLS"/snaps-state is-confinement-supported devmode
+            ;;
+        ubuntu-core-*)
+            not "$TESTSTOOLS"/snaps-state is-confinement-supported classic
+            "$TESTSTOOLS"/snaps-state is-confinement-supported strict
+            "$TESTSTOOLS"/snaps-state is-confinement-supported devmode
+            ;;
+        ubuntu-1*|ubuntu-2*)
+            "$TESTSTOOLS"/snaps-state is-confinement-supported classic
+            "$TESTSTOOLS"/snaps-state is-confinement-supported strict
+            "$TESTSTOOLS"/snaps-state is-confinement-supported devmode
+            ;;
+    esac
+
+    if ls "$SPREAD_PATH"/../snapd_*.deb; then
+        # Check the repack of the snapd deb into core and snapd snaps
+        "$TESTSTOOLS"/snaps-state repack-snapd-deb-into-snap core
+        test -e core/core-from-snapd-deb.snap
```

Comment:
I think we should verify that at least one well know file matches the the source. eg. compare sha1 of /usr/lib/snapd/snapd binary. This will probably need some sort of helper function to do that:
```
sha256_of_snap_file() {
    local snapfile
    local filename
    local tmpdir
    tmpdir=$(mktemp -d)
    echo "$snapfile" > "$tmpdir/extract-file-list"
    unsquashfs -d "$tmpdir/out" -ef "$tmpdir/extract-file-list" "$snapfile"
    local sum
    sum="$(sha256sum "$tmpdir/out/$snapfile")"
    rm -rf "$tmpdir"
    echo "$sum"
}
```

Reply:
Check added